.github/workflows/wiki_notification.yml

```diff
@@ -0,0 +1,45 @@
+name: Wiki Notification
+
+on: [gollum,workflow_dispatch]
+  
+
+jobs:
+  run_script:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: get informations from api and prepare email messages
+      run: |
+        response=$(curl -s https://api.github.com/repos/cgal/cgal/events)
+        eventcount=$(echo "$response" | jq '. | length')
+        for ((i=eventcount; i>=0; i--))
+        do
+            if [ "$(echo "$response" | jq -r '.['$i'].type')" == "GollumEvent" ];
+            then
+                event=$(echo "$response" | jq '.['$i']')
+            fi
+        done
+        actor=$(echo "$event" | jq '.actor.login' | tr -d '"')
+        count=$(echo "$event" | jq  '.payload.pages | length')
+        date=$(echo "$event" | jq '.created_at' | tr -d '"')
+        echo -e "Subject:Cgal wiki has been updated \n" >> messages.txt
+        for ((i=0; i<count; i++)); do
+          action=$(echo "$event" | jq '.payload.pages['$i'].action' | tr -d '"')
+          wiki_page=$(echo "$event" | jq '.payload.pages['$i'].title' | tr -d '"')
+          page_url=$(echo "$event" | jq '.payload.pages['$i'].html_url' | tr -d '"')
+          echo "Page $i" >> messages.txt
+          echo "Actor : $actor" >> messages.txt
+          echo "Action : $action" >> messages.txt
+          echo "Wiki Page : $wiki_page" >> messages.txt
+          echo "Page URL : $page_url" >> messages.txt
+          echo "History : $page_url/_history" >> messages.txt
+          echo "Date : $date" >> messages.txt
+        done
```

Comment:
I am surprise that this code uses `https://api.github.com/repos/cgal/cgal/events` to dig for a matching event, whereas we can get the event that triggered the workflow directly.

See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#gollum and https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#gollum

For example:
```yaml
name: Send notifications on Gollum event

on:
  gollum:
    types: [created, edited, deleted]

jobs:
  send-notification:
    runs-on: ubuntu-latest
    steps:
      - name: Send notification
        uses: actions/github-script@v5
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          script: |
            const eventName = context.eventName;
            const eventPayload = context.payload;
            const repoName = eventPayload.repository.name;
            const sender = eventPayload.sender.login;
            const pages = eventPayload.pages;

            let message = `Gollum event ${eventName} detected in repo ${repoName} by ${sender}. Pages affected:`;
            pages.forEach(page => {
              message += `\n- ${page.page_name}`;
            });
```